pkg/certificate/providers/tresor/fake/fake.go

```diff
@@ -17,14 +18,63 @@ import (
 
 const (
 	rootCertOrganization = "Open Service Mesh Tresor"
+	initialRootName      = "osm-mesh-root-certificate"
 )
 
-type fakeMRCClient struct{}
+type fakeMRCClient struct {
+	mrcChannel chan certificate.MRCEvent
+}
+
+// NewFakeMRC allows for publishing events on to the watch channel to generate MRC events
+func NewFakeMRC() *fakeMRCClient {
+	ch := make(chan certificate.MRCEvent)
 
+	return &fakeMRCClient{
+		mrcChannel: ch,
+	}
+}
+
+// NewCertEvent allows pushing MRC events which can trigger cert changes
+func (c *fakeMRCClient) NewCertEvent(name, state string) {
```

Comment:
This works, alternatively you could leverage a real informer with a fake client, and create new MRC's which would trigger the event.

Don't really have an opinion just wanted to present the alternative :)

Reply:
makes sense, I didn't see the informer usage so didn't introduce it.  As we add more scenarios we may want to go that route